File: plugins/Combat/src/test/java/com/dattilio/scoundrel/CombatParserTest.java
package com.dattilio.scoundrel;

import org.junit.Before;
import org.junit.Test;
import org.mockito.Mock;

import static com.nhaarman.mockito_kotlin.MockitoKt.doNothing;
import static org.mockito.MockitoAnnotations.initMocks;


public class CombatParserTest {
    String string = "[Success: 95, Roll: 91] A pale white rat with crimson splotches misses you with its claws. You dodge a pale white rat with crimson splotches's attack.\n";

    private CombatParser processor;
    CombatPreProcessor mockPresenter;

    @Before
    public void setUp() throws Exception {
        initMocks(this);
        processor = new CombatParser(mockPresenter);
    }

    @Test
    public void updateEngaged() throws Exception {

        processor.processLine(string);
    }

}

File: plugins/Combat/src/test/kotlin/scoundrel/CombatParserTest.kt
package scoundrel

import com.dattilio.scoundrel.CombatParser
import com.dattilio.scoundrel.CombatPreProcessor
import com.nhaarman.mockito_kotlin.doNothing
import com.nhaarman.mockito_kotlin.mock
import com.nhaarman.mockito_kotlin.whenever
import org.junit.Before
import org.junit.Test

class CombatParserTest {
    internal var string = "[Success: 95, Roll: 91] A pale white rat with crimson splotches misses you with its claws. You dodge a pale white rat with crimson splotches's attack.\n"

    internal var mockPresenter: CombatPreProcessor = mock()
    private var processor: CombatParser = CombatParser(mockPresenter)

    @Test
    @Throws(Exception::class)
    fun updateEngaged() {

        processor.parseOpponent(string,true)
    }
}